docs/site/tutorials/todo-list/todo-list-tutorial-sqldb.md

```diff
@@ -24,22 +31,689 @@ rm src/datasources/db.datasource.*
+
+- you already have your own MySQL installation
+- you followed the steps in
+  [Setting up MySQL in Docker](./todo-list-tutorial-docker-mysql.md),
+
+For the purposes of this tutorial, please set the `database` property to
+`testdb`.
+
+The MySQL connector has an extra property, `createDatabase: true`, we can add to
+the datasource settings to create the database for us if it doesn't already
+exist. (If you are using a different relational database, you will need to
+manually create the database before attempting to connect to it.)
+
+The final settings for the datasource for connecting to the Docker MySQL setup:
+
+{% include code-caption.html content="/src/datasources/db.datasource.ts" %}
+
+```json
+const config = {
+  name: 'db',
+  connector: 'mysql',
+  url: '',
+  host: '127.0.0.1',
+  port: 3306,
+  user: 'root',
+  password: 'my-secret-pw',
+  database: 'testdb',
+  createDatabase: true,
+};
+```
+
+See [MySQL connector](https://loopback.io/doc/en/lb4/MySQL-connector.html) for
+detailed information about different settings you can specify here, but what we
+currently have is sufficient.
+
+### Change id property's 'generated' setting to true for all models
+
+Currently the `Todo`, `TodoList`, and `TodoListImage` models have an `id`
+property with a setting of `generated: false` .
+
+```json
+ @property({
+    type: 'number',
+    id: true,
+    generated: false,
+  })
+  id?: number;
+```
+
+We need to change the `id` property's `generated` setting to `true` for all
+these models in order for the create and update CRUD functions to work in a
+MySQL database.
+
+```json
+ @property({
+    type: 'number',
+    id: true,
+    generated: true,
+  })
+  id?: number;
+```
+
+### Creating the database tables
+
+Every LoopBack application contains a `/src/migrate.ts` file.
+
+```ts
+import {TodoListApplication} from './application';
+
+export async function migrate(args: string[]) {
+  const existingSchema = args.includes('--rebuild') ? 'drop' : 'alter';
+  console.log('Migrating schemas (%s existing schema)', existingSchema);
+
+  const app = new TodoListApplication();
+  await app.boot();
+  await app.migrateSchema({existingSchema});
+
+  // Connectors usually keep a pool of opened connections,
+  // this keeps the process running even after all work is done.
+  // We need to exit explicitly.
+  process.exit(0);
+}
+
+migrate(process.argv).catch(err => {
+  console.error('Cannot migrate database schema', err);
+  process.exit(1);
+});
+```
+
+It is used either to create or update your database tables with respect to your
+model file definitions.
+
+All you need to know at the moment, is that it boots your application, loads
+your datasource and model files, and then either creates or updates your tables.
+It can be adjusted to perform extra tasks as well.
+
+For the moment, nothing in it needs to be changed. But to understand it in more
+detail, see
+[Database Migrations](https://loopback.io/doc/en/lb4/Database-migrations.html).
```

Comment:
Maybe we can add the link of [Discovery and auto-migration in MySQL](https://loopback.io/doc/en/lb4/MySQL-connector.html#discovery-and-auto-migration) as well as it is more detailed.